core/commands/generate_repository_partials/repository_parser_core.rb

```diff
@@ -610,4 +617,86 @@ def self.append_latest_version(latest_version, releases, product_name, architect
                                             }))
     end
   end
+
+  def parse_cs_latest_all_platforms(repo_url, yum_key, branch, repo_product_ver)
```

Comment:
This must not be inside the core functionality. Especially with names like `cs_repo`. May be it can be generalized for S3 parsing.